homework/H2_Linear_Algebra.jl

```diff
@@ -731,8 +732,123 @@ md"How does a high/low bandwidth affect the prediction? How does regularization
 You can also implement [more kernel functions](https://en.wikipedia.org/wiki/Positive-definite_kernel#Examples_of_p.d._kernels) and pass them to the `kernel_ridge` keyword-argument `kernel` to see how the prediction changes.
 "
 
+# ╔═╡ 71c2b3dd-3fb4-4f8b-9e19-a043e05ff498
+md"### Exercise 4 – Multiple dispatch on diagonal matrix"
+
+# ╔═╡ c986580b-fe8c-4b68-9fa1-230b429e6de8
+md"""
+### Overview diagonal matrix
+A diagonal matrix is one that has non-zero elements only in the diagonal.
```

Comment:
```suggestion
A diagonal matrix is a square matrix that has non-zero elements only in the diagonal.
```

This detail is relevant for our task.

Reply:
This has been added in the most recent commit: [7e5421a](https://github.com/adrhill/julia-ml-course/pull/18/commits/7e5421a1e7a25d63a064edafa878507e1762959d)